internal/server/instance/drivers/driver_qemu.go

```diff
@@ -5935,8 +5910,27 @@ func (d *qemu) detachDisk(name string) error {
 		return err
 	}
 
+	// Check if it's a special device (we don't store detached state on those).
+	if slices.Contains([]string{"agent:config", "cloud-init:config"}, config["source"]) {
```

Comment:
To me, this would only trigger if the agent is added explicitly in Incus config. It avoids a round of DB transactions leading to an ignored error, though.